x/paychan/msg.go

```diff
@@ -35,8 +38,14 @@ func (m *CreatePaymentChannelMsg) Validate() error {
 	if m.Total == nil || m.Total.IsZero() {
 		return errors.Wrap(errors.ErrInvalidMsg, "invalid total amount")
 	}
-	if m.Timeout <= 0 {
-		return errors.Wrap(errors.ErrInvalidMsg, "invalid timeout value")
+	if m.Timeout == 0 {
```

Comment:
Why not add this to `func (t UnixTime) Validate() error`?

And those where it is not required and just do something like:

```go
if timeout != 0 {
  if err := m.Timeout.Validate(); err != nil {
		return errors.Wrap(err, "invalid timeout value")
  }
}
```

Reply:
We seem to check this separately everywhere and it is easy to overlook one place.

Reply:
`m.Timeout == 0` is a valid time. We are supporting time [from year 0 to 9999](https://github.com/iov-one/weave/blob/master/time.go#L11-L15). 1970 has value 0 in UNIX time, but it is a valid time.
Checking if `timeout == 0` is a custom logic, because we can assume in this functionality that 1970 will always be in the past.

Reply:
You mean, if we later use UnixTime to record birthdays, suddenly the person who was born new years 1970 won't be able to add their account? :laughing: 

Fair enough, but I think we want this check most places and easier to add that somewhere. What about `Validate()` and `ValidateNonZero()`, where the second one does both checks? Or `Validate()` and `ValidateAllowZero()`, where we enforce non-zero by default, and can call the second one when we deal with birthdays?

I prefer the default functions do the right thing when the developer doesn't have a deep understanding of the framework (eventually outside devs will use this... hopefully)

Reply:
Somehow I am not convinced. This is for me the same case as [`Coin.Validate`](https://github.com/iov-one/weave/blob/master/coin/coin.go#L262-L279) which also does not fail if the represented amount is zero. Zero is a valid value. It might be not acceptable by some logic, but it has nothing to do with the type.

I think instead of checking here if `m.Timeout == 0` it is better to check if the timeout value is less than 1514768400000 (which is _Monday, January 1, 2018 1:00:00 AM GTM_).
 
I believe the above is a custom logic and checking if the `Timeout` value is zero is confusing. We don't care if the value is exactly zero. We do care if the value is in the past and in this case we can safely assume that everything before 2018 is in the past.

How about instead of changing `UnitTime`, I will change this code (and related comment) to equivalent of this pseudo code:

```go
// Declared globally at the package level.
var year2018 = time.Parse("2018-01-01")

// Explanation why... 2018 is always in the past
if m.Timeout.Time().Before(year2018) {
    return errors.Wrap(ErrFoo, "timeout must be in the future")
}
```

Reply:
Fair enough.
Do it that way and merge when done